src/runtime/entitlements-manager.js

```diff
@@ -116,6 +116,15 @@ export class EntitlementsManager {
       this.responsePromise_ = this.getEntitlementsFlow_(encryptedDocumentKey);
     }
     return this.responsePromise_.then(response => {
+      if (
+        encryptedDocumentKey != null &&
+        response.decryptedDocumentKey == null
+      ) {
+        // Fail to avoid cases of malicious keys.
+        return Promise.reject(
```

Comment:
Would it be friendlier to 3rd party developers (and users if a 3rd party's server mistakenly didn't return a decrypted key) if we logged a warning or error instead?

Reply:
I think it would be better to fail since we don't want to run an encrypted script on a SwG site since it isn't validated.